pallets/parachain-staking/src/lib.rs

```diff
@@ -2622,81 +2301,22 @@ pub mod pallet {
 			Ok(unstaking_len)
 		}
 
-		/// Process the coinbase rewards for the production of a new block.
-		///
-		/// # <weight>
-		/// Weight: O(1)
-		/// - Reads: Balance
-		/// - Writes: Balance
-		/// # </weight>
-		fn do_reward(who: &T::AccountId, reward: BalanceOf<T>) {
-			// mint
-			if let Ok(imb) = T::Currency::deposit_into_existing(who, reward) {
-				Self::deposit_event(Event::Rewarded(who.clone(), imb.peek()));
-			}
-		}
-
-		/// Annually reduce the reward rates for collators and delegators.
-		///
-		/// # <weight>
-		/// Weight: O(1)
-		/// - Reads: LastRewardReduction, InflationConfig
-		/// - Writes: LastRewardReduction, InflationConfig
-		/// # </weight>
-		fn adjust_reward_rates(now: T::BlockNumber) -> Weight {
-			let year = now / T::BLOCKS_PER_YEAR;
-			let last_update = <LastRewardReduction<T>>::get();
-			if year > last_update {
-				let inflation = <InflationConfig<T>>::get();
-				// collator reward rate decreases by 2% of the previous one per year
-				let c_reward_rate = inflation.collator.reward_rate.annual * Perquintill::from_percent(98);
-				// delegator reward rate should be 6% in 2nd year and 0% afterwards
-				let d_reward_rate = if year == T::BlockNumber::one() {
-					Perquintill::from_percent(6)
-				} else {
-					Perquintill::zero()
-				};
-
-				let new_inflation = InflationInfo::new(
-					T::BLOCKS_PER_YEAR.saturated_into(),
-					inflation.collator.max_rate,
-					c_reward_rate,
-					inflation.delegator.max_rate,
-					d_reward_rate,
-				);
-				<InflationConfig<T>>::put(new_inflation.clone());
-				<LastRewardReduction<T>>::put(year);
-				Self::deposit_event(Event::RoundInflationSet(
-					new_inflation.collator.max_rate,
-					new_inflation.collator.reward_rate.per_block,
-					new_inflation.delegator.max_rate,
-					new_inflation.delegator.reward_rate.per_block,
-				));
-				<T as Config>::WeightInfo::on_initialize_new_year();
-			}
-			T::DbWeight::get().reads(1)
-		}
```

Comment:
because it's too heavy we moved it to a call?

Reply:
This was moved to the `execute_scheduled_reward_change` function which can be called by any origin. However, this call [will fail when it's too early to be called](https://github.com/KILTprotocol/kilt-node/pull/384/files#diff-842251b834245800808bd740184c1e3b18c6af1f95fb1bd4d484420017fdf86eR1752-R1759).

The downside is, that the yearly adjustment has to be manually called and thus requires some kind of notification mechanism, e.g. a calendar entry. The upside is that it does not impose a risk exceeding the PoV size in inherent calls anymore.

Reply:
I moved the update of `InflationInfo` for the two extrinsics (`set_inflation` and `adjust_yearly_inflation_rate`) to a helper function and added [more documentation explaining why this can be called by any signed origin](https://github.com/KILTprotocol/kilt-node/pull/384/commits/f459050e76ebed0c6d142a70d13a9f66730d60e5#diff-842251b834245800808bd740184c1e3b18c6af1f95fb1bd4d484420017fdf86eR1727-R1729).